lib/client-api/src/story_store.ts

```diff
@@ -451,8 +451,8 @@ export default class StoryStore {
     };
     const defaultArgs: Args = Object.entries(
       argTypes as Record<string, { defaultValue: any }>
-    ).reduce((acc, [arg, { defaultValue }]) => {
-      if (defaultValue) acc[arg] = defaultValue;
+    ).reduce((acc, [arg, argProps]) => {
+      if (argProps && argProps.defaultValue) acc[arg] = argProps.defaultValue;
```

Comment:
```suggestion
      if (argProps?.defaultValue) acc[arg] = argProps.defaultValue;
```

Also I think `argType` would be a better name for the variable.

Reply:
Thanks, I updated the branch.

Reply:
Yes, that is what happening per my understanding. You can have a look at this line: https://github.com/storybookjs/storybook/blob/f286ca2e88ce045216b1cc491ac0ed93d52e0f50/lib/client-api/src/inferArgTypes.ts#L32